app/controllers/members/participants_controller.rb

```diff
@@ -104,9 +104,13 @@ def create
       reason_for_spare_message = I18n.t(:participant_no_masters,
                                         scope: @activity_errors_scope,
                                         activity: @activity.name)
-    elsif !@member.freshman? && @activity.is_freshmans?
+    elsif !((@activity.is_freshmans? && @member.freshmans?) || (@activity.is_penultimates? && @member.penultimate?) || (@activity.is_senior? && @member.is_senior?) || (@activity.is_masters? && @member.masters?))
```

Comment:
Dislike this logic even more. The previous method was much cleaner but the issue it had was that it would not allow combinations of the 4 filters. If it was freshmans and seniors only it would fail on either of those criteria but pass the other one but still enroll as an reservist. Haven't tested this logic either so it might not even work

Reply:
The logic looks alright, I think this is quite clear.
Don't you want to remove line 102 to 106 however?